File: backend/src/models/Medicine.js
const mongoose = require('mongoose');

const medicineSchema = new mongoose.Schema(
  {
    name: { type: String, required: true },
    category: { type: String, required: true },
    price: { type: Number, required: true },
    image: { type: String, default: '' },
    stock: { type: Number, default: 0 },
    prescriptionRequired: { type: Boolean, default: false },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Medicine', medicineSchema);

File: flutter_app/lib/models/medicine.dart
class Medicine {
  final String id;
  final String name;
  final String category;
  final double price;
  final String imageUrl;
  final bool prescriptionRequired;

  Medicine({
    required this.id,
    required this.name,
    required this.category,
    required this.price,
    required this.imageUrl,
    required this.prescriptionRequired,
  });

  factory Medicine.fromJson(Map<String, dynamic> json) {
    return Medicine(
      id: json['_id'].toString(),
      name: json['name'] ?? 'Unknown',
      category: json['category'] ?? 'General',
      price: (json['price'] as num?)?.toDouble() ?? 0,
      imageUrl: json['image'] ?? '',
      prescriptionRequired: json['prescriptionRequired'] ?? false,
    );
  }
}
